b2sdk/transfer/outbound/copy_source.py

```diff
@@ -13,8 +13,8 @@
 
 class CopySource(OutboundTransferSource):
     def __init__(self, file_id, offset=0, length=None):
-        if length is None and offset > 0:
-            raise ValueError('Cannot copy with non zero offset and unknown length')
+        if not length and offset > 0:
+            raise ValueError('Cannot copy with non zero offset and unknown or zero length')
```

Comment:
Why?  Can't we copy a zero-length file?

Reply:
We could, but we'd rather not. Uploads are A-class transactions (free), but `b2_copy_file` are C-class transactions (2500/day for free, then $0.004 per 1,000 calls), so since we are not going to save much bandwidth on transfer, it's better to re-upload the "whole" file.

Reply:
This is an emerger code. See my previous comment.